src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1352,6 +1352,21 @@ private void PostExecution()
         {
             ApplyUpdate();
             HandleWarnings();
+            //reset callsite invocation count after run.
+            ResetCallsiteExecutionState();
+        }
+
+        /// <summary>
+        /// Resets Callsites Execution state - does not reset trace data.
+        /// </summary>
+        private void ResetCallsiteExecutionState()
+        {
+            //TODO what about other ds runners?
+            //TODO move to internal function in RuntimeData object.
+            runtimeCore.RuntimeData.CallsiteCache.Values.ToList().ForEach(x =>
+            {
+                x.UpdateCallsiteExecutionState(null, runtimeCore);
```

Comment:
How about simply resetting for each call site just before returning from `DispatchNew`?

Reply:
unfortunately as discussed this does not work.